nominal/sdk.py

```diff
@@ -0,0 +1,508 @@
+        """
+        start_abs = _flexible_time_to_conjure_scout_run_api(start_time)
+        end_abs = _flexible_time_to_conjure_scout_run_api(end_time) if end_time else None
+        datasets = datasets or {}
+        request = scout_run_api.CreateRunRequest(
+            attachments=list(attachment_rids),
+            data_sources={
+                ref_name: scout_run_api.CreateRunDataSource(
+                    data_source=scout_run_api.DataSource(dataset=rid),
+                    series_tags={},
+                    offset=None,  # TODO(alkasm): support per-dataset offsets
+                )
+                for ref_name, rid in datasets.items()
+            },
+            description=description,
+            labels=list(labels),
+            links=[],  # TODO(alkasm): support links
+            properties={} if properties is None else dict(properties),
+            start_time=start_abs,
+            title=title,
+            end_time=end_abs,
+        )
+        response = self._run_client.create_run(self._auth_header, request)
+        return Run._from_conjure(self._auth_header, self._run_client, response)
+
+    def get_run(self, run_rid: str) -> Run:
+        """Retrieve a run."""
+        response = self._run_client.get_run(self._auth_header, run_rid)
+        return Run._from_conjure(self._auth_header, self._run_client, response)
+
+    def _list_runs_paginated(self, request: scout_run_api.SearchRunsRequest) -> Iterable[scout_run_api.Run]:
+        while True:
+            response = self._run_client.search_runs(self._auth_header, request)
+            yield from response.results
+            if response.next_page_token is None:
+                break
+            request = scout_run_api.SearchRunsRequest(
+                page_size=request.page_size,
+                query=request.query,
+                sort=request.sort,
+                next_page_token=response.next_page_token,
+            )
+
+    def _list_runs(self) -> Iterable[Run]:
+        # TODO(alkasm): search filters
+        # TODO(alkasm): put in public API when we decide if we only expose search, or search + list.
+        request = scout_run_api.SearchRunsRequest(
+            page_size=100,
+            query=scout_run_api.SearchQuery(),
+            sort=scout_run_api.SortOptions(
+                field=scout_run_api.SortField.START_TIME,
+                is_descending=True,
+            ),
+        )
+        for run in self._list_runs_paginated(request):
+            yield Run._from_conjure(self._auth_header, self._run_client, run)
+
+    def create_dataset_from_io(
+        self,
+        dataset: BinaryIO,
+        name: str,
+        timestamp_column_name: str,
+        timestamp_column_type: TimestampColumnType,
+        file_extension: _AllowedFileExtensions = ".csv",
+        *,
+        description: str | None = None,
+        labels: Sequence[str] = (),
+        properties: Mapping[str, str] | None = None,
+    ) -> str:
```

Comment:
should this return a `Dataset` to align up with `create_run` which returns a `Run`?

Reply:
Ya I wasn't sure if I should mirror the APIs directly (for runs, the APIs return a run, for datasets, the APIs return a rid) or if I should make additional API calls to make them behave the same.

Reply:
should I rename them back to `update()` then? maybe more intuitive?